fbpic/particles/elementary_process/ionization/ionizer.py

```diff
@@ -156,9 +157,10 @@ def initialize_ADK_parameters( self, element, dt ):
         if self.use_cuda:
             self.prng = PRNG()
 
```

Comment:
The changes below correspond to the reduction of the duplication between the CPU and GPU code. Instead of having two methods (`handle_ionization_gpu` and `handle_ionization_cpu`), I now have a single method (`handle_ionization`), which uses code relevant to GPU or CPU only when needed.